metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -267,6 +267,20 @@ public TableDto deleteAndReturn(final QualifiedName name, final boolean isMView)
         // Try to delete the table even if get above fails
         try {
             connectorTableServiceProxy.delete(name);
+
+            // If this is a common view, the storage_table if present
+            // should also be deleted.
+            if (MetacatUtils.isCommonView(tableDto.getMetadata())
+                    && config.deleteCommonViewStorageTable()) {
+                final Optional<String> storageTableName = MetacatUtils
+                        .getCommonViewStorageTable(tableDto.getMetadata());
+                if (storageTableName.isPresent()) {
+                    final QualifiedName qualifiedStorageTableName = QualifiedName.ofTable(name.getCatalogName(),
+                            name.getDatabaseName(), storageTableName.get());
+                    deleteCommonViewStorageTable(name, qualifiedStorageTableName);
```

Comment:
Construct the `qualifiedStorageTableName` using view name as db which is default.